From 2a3f53494df93d4fe0e67055f95fc3e33cf7e8f4 Mon Sep 17 00:00:00 2001
From: Stephen Finucane <stephenfin@redhat.com>
Date: Mon, 24 Jun 2019 16:07:20 +0100
Subject: [PATCH] Add support for Sphinx v2.1

Come a new Sphinx version, come new breakages. This one is because of
commit 107c20a11f [1], which changed the default centering style of
tables.

[1] https://github.com/sphinx-doc/sphinx/commit/107c20a11f

Change-Id: I1155300828ad0012bd80b7615a0baca121c5e72e
Signed-off-by: Stephen Finucane <stephenfin@redhat.com>
---
 os_api_ref/tests/test_basic_example.py | 15 +++++++++------
 os_api_ref/tests/test_microversions.py |  4 ++--
 2 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/os_api_ref/tests/test_basic_example.py b/os_api_ref/tests/test_basic_example.py
index 20a4fad..dcfb7d7 100644
--- a/os_api_ref/tests/test_basic_example.py
+++ b/os_api_ref/tests/test_basic_example.py
@@ -94,7 +94,7 @@ class TestBasicExample(base.TestCase):
         # TODO(stephenfin): Drop support for this once we drop support for both
         # Python 2.7 and Sphinx < 2.0, likely in "U"
         if sphinx.version_info >= (2, 0, 0):
-            table = """<table class="docutils align-center">
+            table = """<table class="docutils align-{}">
 <colgroup>
 <col style="width: 20%"/>
 <col style="width: 10%"/>
@@ -115,7 +115,8 @@ class TestBasicExample(base.TestCase):
 <td><p>The name of things</p></td>
 </tr>
 </tbody>
-</table>"""
+</table>""".format('center' if sphinx.version_info < (2, 1, 0)
+                   else 'default')
         else:
             table = """<table border="1" class="docutils">
 <colgroup>
@@ -147,7 +148,7 @@ class TestBasicExample(base.TestCase):
         # TODO(stephenfin): Drop support for this once we drop support for both
         # Python 2.7 and Sphinx < 2.0, likely in "U"
         if sphinx.version_info >= (2, 0, 0):
-            success_table = """<table class="docutils align-center">
+            success_table = """<table class="docutils align-{}">
 <colgroup>
 <col style="width: 30%"/>
 <col style="width: 70%"/>
@@ -168,9 +169,10 @@ class TestBasicExample(base.TestCase):
 <td><p>Resource was created and is ready to use.</p></td>
 </tr>
 </tbody>
-</table>"""
+</table>""".format('center' if sphinx.version_info < (2, 1, 0)
+                   else 'default')
 
-            error_table = """<table class="docutils align-center">
+            error_table = """<table class="docutils align-{}">
 <colgroup>
 <col style="width: 30%"/>
 <col style="width: 70%"/>
@@ -200,7 +202,8 @@ class TestBasicExample(base.TestCase):
 <td><p>There is already a zone with this name.</p></td>
 </tr>
 </tbody>
-</table>"""
+</table>""".format('center' if sphinx.version_info < (2, 1, 0)
+                   else 'default')
         else:
             success_table = """table border="1" class="docutils">
 <colgroup>
diff --git a/os_api_ref/tests/test_microversions.py b/os_api_ref/tests/test_microversions.py
index 11f3636..9aa4bf2 100644
--- a/os_api_ref/tests/test_microversions.py
+++ b/os_api_ref/tests/test_microversions.py
@@ -58,7 +58,7 @@ class TestMicroversions(base.TestCase):
         """Test that min / max mv css class attributes are set in params"""
         if sphinx.version_info >= (2, 0, 0):
             table = """<div class="api-detail collapse section" id="list-servers-detail">
-<table class="docutils align-center">
+<table class="docutils align-{}">
 <colgroup>
 <col style="width: 20%"/>
 <col style="width: 10%"/>
@@ -95,7 +95,7 @@ class TestMicroversions(base.TestCase):
 </tbody>
 </table>
 </div>
-"""  # noqa
+""".format('center' if sphinx.version_info < (2, 1, 0) else 'default')  # noqa
         else:
             table = """<div class="api-detail collapse section" id="list-servers-detail">
 <table border="1" class="docutils">
-- 
2.22.0

